app/controllers/candidates_controller.rb

```diff
@@ -69,4 +72,10 @@ def get_candidate
   def candidate_params
     params.require(:candidate).permit(:user_id, :candidate_fullname, :date_of_birth, :phone, :avatar, :cv)
```

Comment:
anh giải thích giúp em về require và permit đc ko ?

Reply:
require là bắt buộc phải tồn tại parameter đó.
còn permit là các key và value được phép pass vào 
ví dụ
`.new(candidate: { candidate_fullname: "Ho Viet Tuan", phone: 016899866 }) // okay
.new(candidate: { candidate_fullname: "Ho Viet Tuan" }) // okay
.new(candidate: { name: "Ho Viet Tuan", other: 'giang bai hay vl' }) // not okay, other not permitted
.new(candidate:{ first_name: "Tuan " }) // not okay, first_name not permitted
.new(employer: { candidate_fullname: "Bhojendra", phone: 016899866 }) // not okay, must be a candidate`